### PR TITLE
ref(dynamic-sampling): Split function for sampling match

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR is a preparation for the refactoring of dynamic sampling. It contains the splitting of the matching function in order to decouple `ProjectState` and `SamplingConfig`.

### CODE REVIEW

File: relay-server/src/utils/dynamic_sampling.rs [hunk 1]
```diff
@@ -50,21 +50,15 @@ fn check_unsupported_rules(
 
 /// Gets the sampling match result by creating the merged configuration and matching it against
 /// the sampling configuration.
-fn get_sampling_match_result(
+pub fn merge_configs_and_match(
```
Comment (iker-barriocanal) [inline relay-server/src/utils/dynamic_sampling.rs]: If you think about the semantical usage of this function, there're two steps that we're doing here. It'd be great if we could split this function into the merging and matching functions. Could we do this in a follow-up PR?

Comment (iker-barriocanal) [inline relay-server/src/utils/dynamic_sampling.rs]: Do we need this to be public?
```suggestion
fn merge_configs_and_match(
```

**Review by iker-barriocanal — APPROVED** [review]
Thank you for taking the time to improve the code quality, and splitting up the original PR!

Please see the comment on the `pub` before merging, and consider continuing the refactor in follow-up PRs.

File: relay-server/src/utils/dynamic_sampling.rs (hunk 1 above)
Comment (iambriccardo) [reply relay-server/src/utils/dynamic_sampling.rs]: Ehm, they are already split up in lines 66 and 67. This function orchestrates their behavior, this is why it is named like this.

Comment (iambriccardo) [reply relay-server/src/utils/dynamic_sampling.rs]: Thanks!

Comment (iker-barriocanal) [reply relay-server/src/utils/dynamic_sampling.rs]: I was thinking something like the following:

```rs
fn get_sampling_match() {
  merge_configs();
  match_configs();
}
```

Comment (iambriccardo) [reply relay-server/src/utils/dynamic_sampling.rs]: I don't think that the complexity of the code requires the creation of a third function encapsulating that behavior. If it was more complex I would have agreed. On the other hand, if the encapsulation follows some strict Relay guidelines I am happy to adapt.

**Pull request merged**